LiteCore/Query/QueryParser.cc

```diff
@@ -320,9 +320,12 @@ namespace litecore {
 
 
     void QueryParser::writeStringLiteralAsProperty(slice str) {
-        require(str.size > 0 && str[0] == '.',
-                "Invalid property name '%.*s'; must start with '.'", SPLAT(str));
-        str.moveStart(1);
+        require(str.size > 0 && (str[0] == '.' || str[0] == '*'),
```

Comment:
Yeah I considered that...will that avoid allocating memory by virtue of the const char * being in the const section of the binary and having the slice just point to it?